### DISCUSSION TITLE
Server Error

### DISCUSSION THREAD

**Discussion by timchen1216** [1101-GS3073-C/Python-Community-GS3073-C#308] [Q&A] [answered]
在畢業專題中，我們用flask做了一個網站，也部署到azure上。但是連過去時卻出現以下畫面
![image](https://user-images.githubusercontent.com/95116299/166453866-5be82bca-7cbb-4297-9848-df7c4a243b15.png)

我們自己在電腦上跑的時候都沒有問題，理論上應該是長這樣
![image](https://user-images.githubusercontent.com/95116299/166454013-cf7a266a-09ce-41d5-b6aa-ad838f1dc509.png)

因為我們網頁的後端有使用到mongodb，所以我們懷疑是azure連不上mongodb，但不知道如何解決
以下是我們的儲存庫
https://github.com/timchen1216/Library_web

**Comment by timchen1216** [comment, marked as answer]
問題已解決

**Reply by EricAgumonJou** [reply]
好奇想問一下後來是如何解決的~(我也曾經碰過類似的狀況)

**Reply by timchen1216** [reply]
原本我是使用mongodb資料庫，這個資料庫是將資料存在雲端上，好處是無論在哪都能取得資料。但是缺點是連線很久，導致常常連線逾時。後來經過我和張家凱教授討論過後，決定改用SQlite。SQlite是將資料存在一個db檔，因此直接讀取檔案會快很多。結論就是如果你的資料是單一形式且資料量比較少的就用SQlite，反之則用mongodb或其他資料庫。
希望以上能對你有幫助d(`･∀･)b
